pkg/operator/ceph/cluster/crash/reconcile.go

```diff
@@ -220,6 +218,25 @@ func (r *ReconcileNode) cephPodList() ([]corev1.Pod, error) {
 	return cephPods, nil
 }
 
+func (r *ReconcileNode) listCrashCollectorAndDelete(nodeName, ns string) error {
```

Comment:
There will just be one crash collector deployment per node with a predictable name, right? Instead of listing, how about if we just call `Delete()` for the deployment directly? If it doesn't exist we just ignore the error.

Reply:
I wanted to have a common method for the code that checks for a stuck crash collector in a terminating state. Yes, in practice we should only have one. To call `Delete()` we need Rook's clientset, unfortunately, the `ReconcileNode` does not have it, only the controller runtime client which requires an object. So unless we refactor a bit we cannot address this.
Because it's a small backport I don't want to change it too much.

How does that sound?

Reply:
The `deleteCrashCollector()` func uses the client like this to delete the deployment by creating a Deployment type, so it seems very doable with the same pattern.
```go
	dep := &appsv1.Deployment{
		ObjectMeta: metav1.ObjectMeta{
			Name:      deploymentName,
			Namespace: namespace,
		},
	}

	err := r.client.Delete(context.TODO(), dep)
```

Reply:
Actually, we need to list the deployments because we don't have the namespace, the reason why the other part of the code works is that even if we pass an empty namespace, the label `client.MatchingLabels{k8sutil.AppAttr: AppName, NodeNameLabel: nodeName}` will find it. Then the returned `&appsv1.DeploymentList{}` has the correct namespace in it. So simple calling `r.client.Delete(context.TODO(), dep)` is not possible since we are missing the namespace in our request.

We cannot pass the namespace name from the request since we are watching Node objects, and they are not namespace. So getting the namespace is not possible. See:

```
NAME                              SHORTNAMES     APIGROUP                              NAMESPACED   KIND
nodes                             no                                                   false        Node
```

Sounds good?

Reply:
Got it, during removal we have no context about the namespace, thanks